server/@types/express/index.d.ts

```diff
@@ -1,28 +1,44 @@
 export declare module 'express-session' {
   // Declare that the session will potentially contain these additional fields
   interface SessionData {
-    user: any
-    identity: any
-    landingPage?: boolean
-    email?: boolean
-    returnTo: string
+    passport?: {
+      user?: Express.User
+    }
+    returnTo?: string
     nowInMinutes: number
   }
 }
 
 export declare global {
   namespace Express {
     interface User {
+      sub: string
+      email?: string
+      email_verified?: boolean
+      phone_number?: string
+      phone_number_verified?: boolean
+      token?: string
       username: string
-      token: string
-      authSource: string
+      authSource: AuthSource
     }
 
+    export type AuthSource = 'external' | 'nomis' | 'delius' | 'azuread'
```

Comment:
Suggestion: Duplicate this line and comment it out. On the new line just add `onelogin` instead of the other values.

`external` means something specific in `hmpps` land. It is likely this application will not need `nomis` or `delius`, but the commented line can act as documentation for you. May be even add a line to state what they are.